### DISCUSSION TITLE
Numeric UpDown, ListboxItem, DragnDrop

### DISCUSSION THREAD

**Discussion by renrutsirhc** [AvaloniaUI/Avalonia#6337] [Q&A]
Hi all,

I have a strange issue which I'm wondering if anyone else has come across:

I have a ListBox. The items template for the listbox contains a NumericUpDown control among other things. I am using a behavior to enable drag and drop reordering the items in the listbox. Here's what happens when I drag one of the buttons on a numeric updown in a listboxitem:

https://user-images.githubusercontent.com/12220071/127726688-e0fa9db3-6ee6-40a9-9968-aa1967970d98.mp4

Has anyone else come accross this behaviour?
Cheers
Chris

**Comment by maxkatz6** [comment]
Is this behavior part of AvaloniaBehaviors? Probably you should create issue there.
@wieslawsoltes

**Reply by renrutsirhc** [reply]
It was based on a combination of the draggable and the standard dragdrop behavior. However, I've just tried removing the behavior completely and I can recreate the same problem with the following in the code behind:

```
 public class SectionSettingsView : UserControl
    {
        private ListBox _Sections;
        public SectionSettingsView()
        {
            InitializeComponent();
            _Sections.PointerMoved += DoDrag;
            AddHandler(DragDrop.DropEvent, Drop);
            AddHandler(DragDrop.DragOverEvent, DragOver);
        }

        private async void DoDrag(object sender, Avalonia.Input.PointerEventArgs e)
        {
            if (e.GetPointerPoint(null).Properties.IsLeftButtonPressed == true)
            {
                DataObject dragData = new DataObject();
                dragData.Set("", new object());
                var result = await DragDrop.DoDragDrop(e, dragData, DragDropEffects.Move);
            }
        }

        private void DragOver(object sender, DragEventArgs e)
        {
        }

        private void Drop(object sender, DragEventArgs e)
        {
        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
            _Sections = this.Find<ListBox>("Sections");
        }
    }
```

It doesn't happen when I remove the call to DoDragDrop from the PointerMoved event. Any ideas?

**Reply by maxkatz6** [reply]
RepeatButton inside of NumericUpDown handles PointerPressed and starts timer to increment its value.

**Reply by renrutsirhc** [reply]
Thanks Max - so can you suggest how could stop this from happening? Possible to override the PointerPressed for the RepeatButton in the NumericUpDown?
